### PR TITLE
Strange Ball & Strange Egg

### PR DESCRIPTION
**Pull request by jknpj**
**STRANGE BALL**

ball2
https://www.webmshare.com/jQaoO (Old webm)

A small metallic ball with **TG17355** carved on its surface. You pick it up and activate its manual boot just like with a posibrain but instead of becoming an annoying talking ball it transforms into some kind of tiny cyborg with this strange **TG17355** module installed.

**STRANGE EGG**

egg2
http://webmshare.com/9nPrq (Old webm)

Some kind of egg-like machine with **TG17355** carved on its surface. After being actived it transforms into a bear-egg cyborg thing with this strange **TG17355** module installed.

The **TG17355** module contains:

**NEW** tools
* **Hugging Module** 
Normal: Has 2 modes - Hug, Bear hug/Head bop
**Hug** mode will shake people if they're lying up and hug if they're up. It makes the target feel better and has the same paracetamol adding effect as a carbon-to-carbon hug.
**Bear** hug has the same shaking effect but unlike the normal hug mode, these hugs fell uncomfortable and don't make the target feel better at all. Targeting the head will make it harmlessy bop the target's head.
Emaged: Adds 2 extra modes - Shocking hug and Crush grip
**Shocking** hug deals 15 burn damage and apply the same kind of stun a normal shock would. It has a cooldown of 2 seconds. Each shock drains 500 power from the user's battery.
**Chush** grip deals 20 brute damage to the target. It has a cooldown of 2 seconds. Each crush drains 300 power from the user's battery.

* **Sonic Harm Prevention Tool**
Normal:
**HUMAN HARM!** This loud siren leave anyone without proper protection in a 9 tiles radius stuttering, dizzy, jittery and confused for 5 seconds and deaf for 1. It has a cooldown of 20 seconds. Each use consumes 1000 power from the user's battery.
Emaged: Changes to a stronger version
**BZZZZZZZZT!** This hellish noise will leave anyone without proper protection in a 6 tiles radius knocked down for 7 seconds, stuttering and jittering for 30, deaf for 10 and will also wake up anyone sleeping. It has a cooldown of 1 minute. Each use consumes 1000 power from the user's battery.

* **Cookie Synthesizer**
Normal:
Makes **cookies**, good carbons get one, that it. It has a cooldown of 15 seconds. Each use consumes 100 power from the user's battery.
Emaged:
Makes **sleepy cookies** loaded with 10 units of chloral hydrate. It has a cooldown of 15 seconds. Each use consumes 100 power from the user's battery.

* **Peace Hypospray**
Normal: Tranquilizer synthesizer
Makes sleep toxin and cryptobiolin. Can hold 5 units of each and takes 20 seconds to synth 5 units. Each injection uses 5 units.
Emaged: Adds a Cyanide synthesizer
Makes cyanide. Can hold 10 units and takes 10 seconds to synth 5 units. Each injection uses 5 units.

**OLD** tools
* Flash
* Flashlight
* Inflatable Dispenser
* Fire Extinguiser
* Medical hud

Strange ball's omoikane sprite is from Europa, strange egg's Peaceborg sprite from TG and the TG17355 module is @JustSumBody's tgstation/tgstation#17355 port salvaged.

**These cyborgs don't drop MMI/Posibrain after being destroyed nor have one to begin with.**

:cl: 
 * tgs: Adds the TG17355 module as an unobtable module, like the syndicate one.
 * rscadd: Adds the strange ball and the strange egg, two TG17355 cyborgs as a rare small artifact to xenoarch.

### CODE REVIEW

File: code/modules/research/xenoarchaeology/finds/finds.dm [hunk 1]
```diff
@@ -525,7 +525,6 @@
 			possible_spawns += /obj/item/clothing/mask/morphing
 			possible_spawns += /obj/item/clothing/mask/morphing/amorphous
 			//possible_spawns += /obj/item/clothing/mask/happy PENDING REWORK
-			//possible_spawns += /obj/item/clothing/mask/stone WHEN I CODE IT
```
Comment (SonixApache) [inline code/modules/research/xenoarchaeology/finds/finds.dm:528]: nice removed line faggot

Comment (BobdaBiscuit) [reply code/modules/research/xenoarchaeology/finds/finds.dm:528]: Already ingame :^)

**Comment by aratherangryduck** [comment]
Are we really getting cookie synth and hug module borgs.

We already have the failures that were and are PEACEKEEPERS and that look like baby blue trash and now you want to add cookie synthing and hugging borgs. Piss off.

Edit:Im a damn idiot who apparently can't read that itsxemoarch shit. Point and laugh, I deserve it.

**Comment by PJB3005** [comment]
Nerd it's a rare xenoarch artifact.

**Comment by Probe1** [comment]
😆

File: code/modules/research/xenoarchaeology/finds/finds.dm (hunk 1 above)
Comment (jknpj) [reply code/modules/research/xenoarchaeology/finds/finds.dm:528]: I ended commiting a non-updated version of the file then fixed it :man_shrugging:

File: code/game/objects/items/robot/robot_items.dm [hunk 2]
```diff
@@ -35,3 +40,171 @@
...
+
+	if(safety == FALSE)
+		user.visible_message("<span class='big danger'>BZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZT</font>")
+		playsound(get_turf(src), 'sound/machines/warning-buzzer.ogg', 130, 3)
+		for(var/mob/living/carbon/M in hearers(6, user))
+			if(ishuman(M))
+				var/mob/living/carbon/human/H = M
+				if(istype(H.ears, /obj/item/clothing/ears/earmuffs))
+					continue
+			M.sleeping = 0
```
Comment (ghost) [inline code/game/objects/items/robot/robot_items.dm]: One helluva alarm clock

**Comment by silicons** [comment]
Nice my joke module I coded to appease/piss off the security players on tg is getting on VG
Wew

**Comment by silicons** [comment]
I could port the modules over if you want anyways.

File: code/game/objects/items/robot/robot_items.dm [hunk 3]
```diff
@@ -35,3 +40,171 @@
...
+		add_gamelogs(user, "used a [src]", admin = TRUE, tp_link = TRUE, tp_link_short = FALSE, span_class = "notice")
+		return
+
+	if(safety == FALSE)
+		user.visible_message("<span class='big danger'>BZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZT</font>")
+		playsound(get_turf(src), 'sound/machines/warning-buzzer.ogg', 130, 3)
+		for(var/mob/living/carbon/M in hearers(6, user))
+			if(ishuman(M))
+				var/mob/living/carbon/human/H = M
+				if(istype(H.ears, /obj/item/clothing/ears/earmuffs))
```
Comment (JustSumBody) [inline code/game/objects/items/robot/robot_items.dm]: Isn't there an earprot proc for this

**Comment by RemieRichards** [comment]
Awww, why aren't you using the real eggborg sprites ;_;

**Comment by jknpj** [comment]
They are too fat to be handheld. Maybe i can fit them in later as the strange egg

@kevinz000 you're free to port those holosign/barriers if you want

**Comment by RemieRichards** [comment]
Does this ball not spawn an actual borg? cos I meant for the borg, you're right eggborgs are too big to hold, they hold you!

**Comment by jknpj** [comment]
They do spawn an actual borg https://www.webmshare.com/jQaoO

**Comment by RemieRichards** [comment]
yeah that's what I meant about using the egg lol.

**Comment by Kitfox88** [comment]
wanna pat that orb

File: code/modules/reagents/reagent_containers/borghydro.dm [hunk 4]
```diff
@@ -108,13 +108,14 @@
 
 /obj/item/weapon/reagent_containers/borghypo/peace
 	name = "Peace Hypospray"
+	desc = "An tranquilizer synthesizer and injection system. These drugs are capable of inducing a state of relaxation, or euphoria."
```
Comment (ghost) [inline code/modules/reagents/reagent_containers/borghydro.dm]: Shouldn't it be 'A tranquilizer synthesizer and injec...'?

File: code/modules/research/xenoarchaeology/finds/finds_strangeball.dm [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+	name = "strange egg"
+	desc = "A complex egg-like machine with \"TG17355\" carved on its surface."
+	ball = "peaceegg"
+	burger = "peaceborg"
+	icon_state = "peaceegg"
+	w_class = W_CLASS_GIANT
+	density = 1
+
+/obj/item/device/mmi/posibrain/strangeball/strangeegg/attack_hand(mob/user)
+	if(ishuman(user) && user:gloves)
```
Comment (Shadowmech88) [inline code/modules/research/xenoarchaeology/finds/finds_strangeball.dm]: I'm pretty sure use of the `:` operator is frowned upon.

Comment (jknpj) [reply code/modules/research/xenoarchaeology/finds/finds_strangeball.dm]: i just mindlessly copypasted it from artifact code, don't worry i'll remove it.

**Comment by Blithering** [comment]
These things are small artifacts? Might be time to stop rushing for larges.

**Comment by Plaguewalker** [comment]
Reminds of the damn gordys ball from borderlands.

**Comment by optimumtact** [comment]
typeless loop

**Comment by Kitfox88** [comment]
>Reminds of the damn gordys ball from borderlands.

gordys was precious

**Comment by Probe1** [comment]
Why didn't Travis compile..

**Pull request merged**